### PR TITLE
Update hooks for farms/syrups

### PR DESCRIPTION
**Pull request by totop716**
- Remove console errors when the data is not fetched from multicall
- Update farm/syrup card according to hook update

### CODE REVIEW

File: src/components/FarmCard/FarmCardDetails.tsx [hunk 1]
```diff
@@ -503,7 +503,7 @@ const FarmCardDetails: React.FC<{
                   !attemptUnstaking &&
                   Number(unstakeAmount) > 0 &&
                   Number(unstakeAmount) <=
-                    Number(stakingInfo.stakedAmount.toExact())
+                    Number(stakingInfo.stakedAmount?.toExact() ?? 0)
```
Comment (FriskyHamTitz) [inline src/components/FarmCard/FarmCardDetails.tsx:506]: Why aren't we using proper typings, amounts should always be a number

File: src/components/StakeSyrupModal/StakeSyrupModal.tsx [hunk 2]
```diff
@@ -57,29 +57,31 @@ const StakeSyrupModal: React.FC<StakeSyrupModalProps> = ({
...
   );
 
   const parsedAmountWrapped = wrappedCurrencyAmount(parsedAmount, chainId);
 
-  let hypotheticalRewardRate: TokenAmount = new TokenAmount(
-    syrup.rewardRate.token,
-    '0',
-  );
+  let hypotheticalRewardRate = syrup.rewardRate
+    ? new TokenAmount(syrup.rewardRate.token, '0')
```
Comment (FriskyHamTitz) [inline src/components/StakeSyrupModal/StakeSyrupModal.tsx:75]: Why are we passing 0 as a string?

File: src/components/SyrupCard/SyrupCard.tsx [hunk 3]
```diff
@@ -101,15 +97,12 @@ const SyrupCard: React.FC<{ syrup: SyrupInfo; dQUICKAPY: string }> = ({
...
-                        syrup.rate * Number(syrup.rewardTokenPriceinUSD)
-                      ).toFixed(2),
-                    ).toLocaleString()}{' '}
-                    <span style={{ color: palette.text.secondary }}>
-                      {' '}
-                      / day
-                    </span>
+                    {syrup.rewardTokenPriceinUSD
+                      ? (
+                          syrup.rate * Number(syrup.rewardTokenPriceinUSD)
```
Comment (FriskyHamTitz) [inline src/components/SyrupCard/SyrupCard.tsx:102]: Probably not in this PR but in the future we should figure out a way to remove casting to numbers all the time

File: src/pages/DragonPage/DragonsSyrup.tsx [hunk 4]
```diff
@@ -100,9 +100,11 @@ const DragonsSyrup: React.FC = () => {
   const sortByEarned = useCallback(
     (a: SyrupInfo, b: SyrupInfo) => {
       const earnedUSDA =
-        Number(a.earnedAmount.toExact()) * Number(a.rewardTokenPriceinUSD ?? 0);
+        Number(a.earnedAmount?.toExact() ?? 0) *
```
Comment (FriskyHamTitz) [inline src/pages/DragonPage/DragonsSyrup.tsx:103]: Create a prototype extensions method exactOraDefault, so we can hide ?? 0

File: src/state/stake/hooks.ts [hunk 5]
```diff
@@ -448,29 +428,36 @@ export function useSyrupInfo(
...
-          const totalStakedAmount = new TokenAmount(
-            lp && lp !== ''
-              ? new Token(137, lp, 18, 'SLP', 'Staked LP')
-              : syrupInfo.stakingToken,
-            JSBI.BigInt(totalSupplyState.result?.[0]),
-          );
+          const stakedAmount =
+            balanceState?.result && balanceState?.result?.[0]
+              ? new TokenAmount(
+                  lp && lp !== ''
```
Comment (FriskyHamTitz) [inline src/state/stake/hooks.ts:434]: Why aren't we using a constructor or common initializer

File: src/state/stake/hooks.ts [hunk 6]
```diff
@@ -448,29 +428,36 @@ export function useSyrupInfo(
...
+                    : syrupInfo.stakingToken,
+                  JSBI.BigInt(balanceState?.result?.[0] ?? 0),
+                )
+              : undefined;
+          const totalStakedAmount =
+            totalSupplyState.result && totalSupplyState.result?.[0]
+              ? new TokenAmount(
+                  lp && lp !== ''
+                    ? new Token(137, lp, 18, 'SLP', 'Staked LP')
+                    : syrupInfo.stakingToken,
```
Comment (FriskyHamTitz) [inline src/state/stake/hooks.ts:445]: Reuse initializer

File: src/state/stake/hooks.ts [hunk 7]
```diff
@@ -643,38 +624,32 @@ export function useOldSyrupInfo(
...
-          const totalStakedAmount = new TokenAmount(
-            lp && lp !== ''
-              ? new Token(137, lp, 18, 'SLP', 'Staked LP')
-              : syrupInfo.stakingToken,
-            JSBI.BigInt(totalSupplyState.result?.[0]),
-          );
+          const stakedAmount =
+            balanceState?.result && balanceState?.result?.[0]
+              ? new TokenAmount(
+                  lp && lp !== ''
```
Comment (FriskyHamTitz) [inline src/state/stake/hooks.ts:638]: Fix token amount to be in an initializer

File: src/state/stake/hooks.ts [hunk 8]
```diff
@@ -1716,25 +1660,32 @@ export function useOldStakingInfo(
...
-            lp && lp !== ''
-              ? new Token(137, lp, 18, 'SLP', 'Staked LP')
-              : dummyPair.liquidityToken,
-            JSBI.BigInt(totalSupplyState.result?.[0]),
-          );
+          const stakedAmount =
+            balanceState?.result && balanceState?.result?.[0]
+              ? new TokenAmount(
+                  lp && lp !== ''
+                    ? new Token(137, lp, 18, 'SLP', 'Staked LP')
```
Comment (FriskyHamTitz) [inline src/state/stake/hooks.ts:1667]: Fix duplicate code for the whole file

**Review by FriskyHamTitz — CHANGES_REQUESTED** [review]

File: src/components/StakeSyrupModal/StakeSyrupModal.tsx (hunk 2 above)
Comment (totop716) [reply src/components/StakeSyrupModal/StakeSyrupModal.tsx:75]: This is for showing reward token as 0 when the value of staking input is 0.

File: src/state/stake/hooks.ts [hunk 9]
```diff
@@ -1692,49 +1444,24 @@ export function useOldStakingInfo(
...
           // always need these
           totalSupplyState &&
-          !totalSupplyState.loading &&
-          periodFinishState &&
-          !periodFinishState.loading
+          !totalSupplyState.loading
         ) {
-          if (
-            balanceState?.error ||
-            earnedAmountState?.error ||
```
Comment (FriskyHamTitz) [inline src/state/stake/hooks.ts:1701]: What replaces all of this code that was removed?

**Review by FriskyHamTitz — APPROVED** [review]
I'm still a bit confused on why we always need to convert amounts to numbers.

If there's not a good reason we should change this in a seperate task

File: src/state/stake/hooks.ts (hunk 9 above)
Comment (totop716) [reply src/state/stake/hooks.ts:1701]: It was returning many console errors because of this.

**Pull request merged**